plat/common/x86/acpi.c

```diff
@@ -136,6 +137,38 @@ static void acpi_list_tables(void)
 }
 #endif /* UK_DEBUG */
 
+static acpi_rsdp_t *acpi_get_efi_st_rsdp(void)
+{
+	struct ukplat_bootinfo *bi = ukplat_bootinfo_get();
+	uk_efi_uintn_t ct_count, i;
+	uk_efi_cfg_tab_t *ct;
+	acpi_rsdp_t *rsdp;
+
+	UK_ASSERT(bi);
+
+	if (!bi->efi_st)
+		return NULL;
+
+	ct = ((uk_efi_sys_tab_t *)bi->efi_st)->configuration_table;
+	ct_count = ((uk_efi_sys_tab_t *)bi->efi_st)->number_of_table_entries;
+
+	UK_ASSERT(ct);
+	UK_ASSERT(ct_count);
+
+	rsdp = NULL;
+	for (i = 0; i < ct_count; i++)
+		if (!memcmp(&ct[i].vendor_guid, UK_EFI_ACPI20_TABLE_GUID,
+			    sizeof(ct[i].vendor_guid)))
+			return ct[i].vendor_table;
```

Comment:
Oh, I have just noticed now that this should have been a `break;` statement instead... otherwise the `uk_pr_debug` below ends up being ignored. :man_facepalming: